test/DynamoCoreTests/SerializationTests.cs

```diff
@@ -17,102 +17,71 @@
-                {
-                    Console.WriteLine(e.Message);
-                }
+                modelsGuidToIdMap.Add(nodeId, idcount.ToString());
+                json = json.Replace(nodeId.ToString("N"), idcount.ToString());
+                idcount = idcount + 1;
             }
 
-            if (Directory.Exists(jsonNonGuidFolder))
+            //alter the output json so that all port ids are not guids
+            foreach (var node in model.Nodes)
             {
-                try
+                foreach (var port in node.InPorts)
                 {
-                    Console.WriteLine("Deleting jsonNonGuid directory from temp");
-                    Directory.Delete(jsonNonGuidFolder, true);
+                    modelsGuidToIdMap.Add(port.GUID, idcount.ToString());
+                    json = json.Replace(port.GUID.ToString("N"), idcount.ToString());
+                    idcount = idcount + 1;
                 }
-                catch (Exception e)
+
+                foreach (var port in node.OutPorts)
                 {
-                    Console.WriteLine(e.Message);
+                    modelsGuidToIdMap.Add(port.GUID, idcount.ToString());
+                    json = json.Replace(port.GUID.ToString("N"), idcount.ToString());
+                    idcount = idcount + 1;
                 }
             }
-        }
-
-        [TestFixtureTearDown]
-        public void TearDown()
-        {
-            ExecutionEvents.GraphPostExecution -= ExecutionEvents_GraphPostExecution;
-        }
-
-        private void ExecutionEvents_GraphPostExecution(Session.IExecutionSession session)
-        {
-            lastExecutionDuration = (TimeSpan)session.GetParameterValue(Session.ParameterKeys.LastExecutionDuration);
-        }
-
-        internal class PortComparisonData
-        {
-            public string ID { get; set; }
-            public bool UseLevels { get; set; }
-            public bool KeepListStructure { get; set; }
-            public int Level { get; set; }
-            public bool UsingDefaultValue { get; set; }
-
-            public override bool Equals(object obj)
+            //alter the output json so that all connectorModel ids are not guids
+            foreach (var connector in model.Connectors)
             {
-                var other = (obj as PortComparisonData);
-                return ID == other.ID &&
-                    other.KeepListStructure == this.KeepListStructure &&
-                    other.Level == this.Level &&
-                    other.UseLevels == this.UseLevels &&
-                    other.UsingDefaultValue == this.UsingDefaultValue;
+                modelsGuidToIdMap.Add(connector.GUID, idcount.ToString());
+                json = json.Replace(connector.GUID.ToString("N"), idcount.ToString());
+                idcount = idcount + 1;
+            }
+            //alter the output json so that all annotationModel ids are not guids
+            foreach (var annotation in model.Annotations)
```

Comment:
Some temp code for note block needed before it's merged into annotation?

Reply:
addressed this in the `doWorkspaceViewCompareAndSave` function